ee/server/service/devices.go

```diff
@@ -16,6 +18,43 @@ func (svc *Service) RequestEncryptionKeyRotation(ctx context.Context, hostID uin
 	return svc.ds.SetDiskEncryptionResetStatus(ctx, hostID, true)
 }
 
+type migrateMDMDevicePayload struct {
+	Timestamp time.Time `json:"timestamp"`
+	Host      struct {
+		ID             uint   `json:"id"`
+		UUID           string `json:"uuid"`
+		HardwareSerial string `json:"hardware_serial"`
+	} `json:"host"`
+}
+
+func (svc *Service) TriggerMigrateMDMDevice(ctx context.Context, host *fleet.Host) error {
+	ac, err := svc.ds.AppConfig(ctx)
+	if err != nil {
+		return err
+	}
+	if !ac.MDM.EnabledAndConfigured {
+		return fleet.NewMDMNotConfiguredError()
+	}
+	if !ac.MDM.MacOSMigration.Enable {
+		return fleet.NewBadGatewayError("macos_migration", ctxerr.New(ctx, "not enabled"))
+	}
+	if ac.MDM.MacOSMigration.WebhookURL == "" {
+		return fleet.NewBadGatewayError("macos_migration", ctxerr.New(ctx, "no webhook URL configured"))
+	}
+
+	p := migrateMDMDevicePayload{}
+	p.Timestamp = time.Now().UTC()
+	p.Host.ID = host.ID
+	p.Host.UUID = host.UUID
+	p.Host.HardwareSerial = host.HardwareSerial
+
+	if err := server.PostJSONWithTimeout(ctx, ac.MDM.MacOSMigration.WebhookURL, p); err != nil {
+		return err
```

Comment:
Might be worth adding `ctxerr.Wrap(...)` here, as it isn't wrapped in `server.PostJSONWithTimeout`.